Updating cognitiveservices-luis-runtime SDK

ahmedabuamra opened the pull request:
All lines of code in this PR are auto-generated except updating the README.md file.

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -17,54 +17,52 @@ npm install @azure/cognitiveservices-luis-runtime
+import { CognitiveServicesCredentials } from "@azure/ms-rest-azure-js";
+import { LUISRuntimeClient } from "@azure/cognitiveservices-luis-runtime";
+
+let subscriptionId = "<luis-authoring-key>";
```
Is `subscriptionId` the right name to use here?
I know that all Cognitive Services will be using the `CognitiveServicesCredentials` which in turn refers to this as `subscriptionKey`.

Which of the 2 makes more sense?

ramya-rao-a replied:
@chrishmsft What are your thoughts here?

ChrisHMSFT replied:
I'd suggest 'Subscription Key' to be consistent with all of the other Cognitive Services. We do call this key the 'authoring key' in luis.ai portal so both will be a little inconsistent.

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -17,54 +17,52 @@ npm install @azure/cognitiveservices-luis-runtime
+
+let subscriptionId = "<luis-authoring-key>";
+const creds = new CognitiveServicesCredentials(subscriptionId);
+const client = new LUISRuntimeClient(creds, "https://westus.api.cognitive.microsoft.com/");
```
Should we add a comment here for user to replace with the endpoint that corresponds to their region?

ahmedabuamra replied:
Yes, I added the comment.

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -17,54 +17,52 @@ npm install @azure/cognitiveservices-luis-runtime
+const creds = new CognitiveServicesCredentials(subscriptionId);
+const client = new LUISRuntimeClient(creds, "https://westus.api.cognitive.microsoft.com/");
+const appId = "ec7b1657-199d-4d8a-bbb2-89a11a42e02a"; // replace this with your appId.
+const versionId = "0.1"; // This is the first versionId, replace it with yours.
```
What does "first versionId" mean?

ahmedabuamra replied:
I mean the initial versionId. By default when you create a luis application, the versionId will be 0.1.

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -145,4 +143,4 @@ See https://github.com/Azure/ms-rest-browserauth to learn how to authenticate to
 - [Microsoft Azure SDK for Javascript](https://github.com/Azure/azure-sdk-for-js)
 
-![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-js/sdk/cognitiveservices/cognitiveservices-luis-runtime/README.png)
+![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-js/sdk/README.png)
```
Did this change come from the auto-generated code?

ahmedabuamra replied:
It's auto-generated. all codes except in the README are auto-generated.

ramya-rao-a replied:
@scbedd Looping you in here as you last worked on impressions. I believe this change is not what we want. Can you confirm?

ramya-rao-a replied:
I have talked to @scbedd offline.
This needs to be fixed in the code generator, but meanwhile for now, please manually change this line to match what is in the master branch 

```suggestion
![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-js/sdk/cognitiveservices/cognitiveservices-luis-runtime/README.png)
```

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/package.json:
```diff
@@ -4,8 +4,11 @@
   "dependencies": {
-    "@azure/ms-rest-js": "^1.8.1",
-    "tslib": "^1.9.3"
+    "@azure/ms-rest-azure-js": "^2.0.1",
```
Why is `@azure/ms-rest-azure-js` being added as a dependency?

ahmedabuamra replied:
Because the tests will need it to authenticate as we have the tests and the sdk in one project.

ramya-rao-a replied:
If only the tests need it, then it should be moved under `devDependencies`.
Otherwise, `@azure/ms-rest-azure-js` will get shipped as part of the luis library to the user

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/src/lUISRuntimeClient.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Can you check why the copyright went missing?
There is a flag `license-header` which you might want to use when running the code generator

ahmedabuamra replied:
I'll regenerate using this flag.

ahmedabuamra replied:
`license-header` flag not working as autorest says can't convert object to string.
Does the order of the flags matter?

ramya-rao-a replied:
I believe the order matters, but not sure.
@michaeljqzq Can you share how you used the code generator for management libraries to be able to get the Copyright info?

ramya-rao-a replied:
Try passing `--license-header=MICROSOFT_MIT_NO_VERSION` instead of just `license-header`

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -17,54 +17,52 @@ npm install @azure/cognitiveservices-luis-runtime
+let subscriptionId = "<luis-authoring-key>";
+const creds = new CognitiveServicesCredentials(subscriptionId);
+const client = new LUISRuntimeClient(creds, "https://westus.api.cognitive.microsoft.com/");
+const appId = "ec7b1657-199d-4d8a-bbb2-89a11a42e02a"; // replace this with your appId.
```
```suggestion
const appId = "<your-app-id>";
```

ramya-rao-a commented:
Please also update the browser sample in the readme

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -17,54 +17,57 @@ npm install @azure/cognitiveservices-luis-runtime
+const client = new LUISRuntimeClient(creds, "https://westus.api.cognitive.microsoft.com/");
+
+const appId = "<your-app-id>"; // replace this with your appId.
+const versionId = "0.1"; // This is the default version ID after creating new luis application.
```
Does the below make sense?

```suggestion
const versionId = "0.1"; // Version of your luis application. Initial value will be 0.1
```

ramya-rao-a commented:
We have 3 main items pending
- The missing copyright
- Impression changing in the readme
- subscription id vs subscription key vs authoring key

I'll figure out the 2nd & 3rd one from my end.
For the first issue, can you add the copyright text back manually until we figure out how to use the code generator to add  it?

ahmedabuamra commented:
What I changed:

- Added the copyrights manually.
- Moved `@azure/ms-rest-azure-js` to `devDependencies`.
- Changed the `Impression ` in README manually.
- Replaced the comment on `versionId` with yours.
- Replaced `subscriptionId` with `authoringKey` as @ChrisHMSFT  suggested.

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -17,54 +17,59 @@ npm install @azure/cognitiveservices-luis-runtime
+const client = new LUISRuntimeClient(creds, "https://"+ region +".api.cognitive.microsoft.com/");
+
+const appId = "<your-app-id>"; // replace this with your appId.
+const versionId = "0.1"; // replace with version of your luis application. Initial value will be 0.1
```
Please make sure to update the browser sample as well

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/package.json:
```diff
@@ -4,8 +4,10 @@
-    "tslib": "^1.9.3"
+    "@azure/ms-rest-js": "^2.0.3",
+    "@types/chai": "^4.2.0",
+    "@types/mocha": "^5.2.7",
```
The 2 `@types` dependencies above are related to testing and so should be under `devDependencies`

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/package.json:
```diff
@@ -19,15 +21,17 @@
+    "uglify-js": "^3.4.9"
   },
-  "homepage": "https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/cognitiveservices/cognitiveservices-luis-runtime",
+  "homepage": "https://github.com/Azure/azure-sdk-for-js",
```
```suggestion
  "homepage": "https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/cognitiveservices/cognitiveservices-luis-runtime",
```

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/rollup.config.js:
```diff
@@ -21,9 +21,6 @@ const config = {
     banner: `/*
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for license information.
- *
```
Please add this copyright text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/src/operations/index.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copyright text back manually

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/src/operations/predictionOperations.ts:
```diff
@@ -1,8 +1,4 @@
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- *
```
Please add the copyright text back manually

ahmedabuamra replied:
I am sorry for missing all of that

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/package.json:
```diff
@@ -4,8 +4,8 @@
   "description": "LUISRuntimeClient Library with typescript type definitions for node.js and browser.",
   "version": "3.0.1",
```
```suggestion
  "version": "4.0.0",
```

ramya-rao-a commented:
Since we have decided to get an update published with this PR approved, can we also get the browser sample updated?

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -17,124 +17,143 @@ npm install @azure/cognitiveservices-luis-runtime
+      // check the following link to find your region
+      // https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-regions
+      const region = "<your-region>";
+      const client = new LUISRuntimeClient(
```
```suggestion
      const client = new Azure.CognitiveservicesLuisRuntime.LUISRuntimeClient(
```

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -17,124 +17,143 @@ npm install @azure/cognitiveservices-luis-runtime
-          },
-          externalEntities: [{
+      let authoringKey = process.env["luis-authoring-key"];
+      const creds = new CognitiveServicesCredentials(authoringKey);
```
Since the `@azure/ms-rest-azure-js` package does not have a browser bundle yet, we should use the `ApiKeyCredentials` from `@azure/ms-rest-js` instead.

```suggestion
      const creds = new ApiKeyCredentials({ inHeader: { 'Ocp-Apim-Subscription-Key': authoringKey } });
```

ramya-rao-a commented on sdk/cognitiveservices/cognitiveservices-luis-runtime/README.md:
```diff
@@ -17,124 +17,143 @@ npm install @azure/cognitiveservices-luis-runtime
     <title>@azure/cognitiveservices-luis-runtime sample</title>
-    <script src="node_modules/@azure/ms-rest-js/dist/msRest.browser.js"></script>
-    <script src="node_modules/@azure/ms-rest-browserauth/dist/msAuth.js"></script>
+    <script src="node_modules/@azure/ms-rest-azure-js/dist/cognitiveServicesCredentials.js"></script>
```
```suggestion
    <script src="node_modules/@azure/ms-rest-js/dist/msRest.browser.js"></script>
```

The `@azure/ms-rest-azure-js` package does not have a browser bundle and I have logged https://github.com/Azure/ms-rest-azure-js/issues/126 to track this issue.

Until that issue is fixed, we can use the `ApiKeyCredentials` from `@azure/ms-rest-js` instead.
